### PR TITLE
Nav-Unification: fix mobile submenus UX

### LINKED ISSUE #48709: Nav Unification: Expanding menus are hard to interact with on mobile
Expanding menus are hard to interact with on mobile in calypso. Clicking on the top level menu item will take you to that section, rather than opening the child menu items. While on wp-admin, works correctly.

image

props @mrfoxtalbot

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request

* closes all sections only for > 782px
* leaves sections open in mobile < 782px

#### Testing instructions

* enable nav-unification (paYJgx-1af-p2)
* on calypso live, switch to a mobile breakpoint (<=660px), try navigating to a submenu. 
* clicking on a submenu header should not redirect to the respective section, it should only expand the submenu

Fixes #48709

### CODE REVIEW

**Review by mmtr — COMMENTED** [review]
Not sure why, but there are certain links that are not expanded. See the GIF below and note how the "Posts" section is expanded successfully but the "Links" one does not.

Jan-14-2021 15-36-26

**Comment by cpapazoglou** [comment]
> Not sure why, but there are certain links that are not expanded. See the GIF below and note how the "Posts" section is expanded successfully but the "Links" one does not.
> 
> Jan-14-2021 15-36-26

Good point @mmtr! 83f1262 should fix it!

**Review by getdave — CHANGES_REQUESTED** [review]
Thanks for working on this. Having been in and around this bit fo the code before I know it can be a frustrating experience so 👍 for persevering here.

Unfortunately, I tried this out locallty and hit a few snags.

I think we need a dedicated variable or setting that determines when the menu becomes "expandable" vs permanently visible. We should then use this setting to determine how the click on the top-level items behaves.

#### Unclear which section is "current"

Once you click on a few sections it becomes unclear which one you are currently on as they all look the same (ie: expanded looks the same as "current").

Screen Shot 2021-01-15 at 16 24 11

#### Navigating with slide out menu

On screens where the menu is hidden and slides in/out we are still using the "click to navigate" method instead of the "click to expand". This causes a frustrating UX when I want to navigate to sub menu item.

Screen Capture on 2021-01-15 at 16-25-21

**Comment by cpapazoglou** [comment]
Thanks @getdave for your thorough review! 

> Once you click on a few sections it becomes unclear which one you are currently on as they all look the same (ie: expanded looks the same as "current").

It seems that this is a color-scheme problem, in particular the following color-schemes have the same color for `--color-sidebar-menu-selected-background = --color-sidebar-menu-hover-background` and `--color-sidebar-menu-selected-text = --color-sidebar-menu-hover-text`:  Blue, Ocean, Coffee, Ectoplasm, Light, Midnight, Modern, Sunrise. With all of them (except modern) being wp-admin originating color-schemes, it means that we need to establish different colors for hovering both in calypso and wp-admin for these color-schemes. Should we follow up in another PR?

The rest color-schemes (eg dark) are like below:


> On screens where the menu is hidden and slides in/out we are still using the "click to navigate" method instead of the "click to expand". This causes a frustrating UX when I want to navigate to sub menu item.

Very good point! Addressed here 53ae2d2

File: client/my-sites/sidebar-unified/menu.jsx [hunk 1]
```diff
@@ -64,18 +64,24 @@ export const MySitesSidebarUnifiedMenu = ( {
 		<li>
 			<ExpandableSidebarMenu
 				onClick={ () => {
-					if ( link ) {
+					// @automattic/viewport doesn't have a breakpoint for medium (782px)
+					if ( window.innerWidth > 782 ) {
```
Comment (getdave) [inline client/my-sites/sidebar-unified/menu.jsx]: I _think_ we should be looking to use [`matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) with MQ string here.

File: client/my-sites/sidebar-unified/menu.jsx [hunk 2]
```diff
@@ -64,18 +64,24 @@ export const MySitesSidebarUnifiedMenu = ( {
 		<li>
 			<ExpandableSidebarMenu
 				onClick={ () => {
-					if ( link ) {
+					// @automattic/viewport doesn't have a breakpoint for medium (782px)
```
Comment (getdave) [inline client/my-sites/sidebar-unified/menu.jsx]: I still think this `782` would be better represented as `$collapsedSidebarViewportWidth` or some such. Self documenting code style.

**Review by getdave — CHANGES_REQUESTED** [review]
Thanks for progress here. It's a tough one.

I tested again and I'm not getting a consistent experience. Let me know what you think.

## Distinguish between "toggled open" and "current

Still finding it difficult to distinguish between "toggled open" and "current" on the smaller viewports.

Screen Shot 2021-01-19 at 15 06 18

## Multiple "selected" items

I got into this state by:

1. Navigating to `Media`.
2. Toggling menu open.
3. Clicking `Links`.
4. Clicking `Posts`
5. Clicking `Upgrades`.

Screen Shot 2021-01-19 at 15 08 22

## Inconsistent toggle experience

Clicking on `Pages` toggles the menu but does not navigate. This is correct.

Clicking on `Feedback` does not toggle the menu but navigates. 

Ditto for `Users`.

This is inconsistent.

File: client/my-sites/sidebar-unified/menu.jsx (hunk 2 above)
Comment (cpapazoglou) [reply client/my-sites/sidebar-unified/menu.jsx]: Since `782` breakpoint is not something temporary, I added it in the viewport package and used `isWithinBreakpoint( '>782px' )` in order to align with the rest codebase.  acf3791 cb17657

File: client/my-sites/sidebar-unified/menu.jsx (hunk 1 above)
Comment (cpapazoglou) [reply client/my-sites/sidebar-unified/menu.jsx]: Since `782` breakpoint is not something temporary, I added it in the viewport package and used `isWithinBreakpoint( '>782px' )` in order to align with the rest codebase.  acf3791 cb17657

**Comment by cpapazoglou** [comment]
Rebased.

Also, @getdave can you confirm that you have read my comment https://github.com/Automattic/wp-calypso/pull/48803#issuecomment-762134998 prior to yours https://github.com/Automattic/wp-calypso/pull/48803#pullrequestreview-571329135?

TL;DR;

The following are color-scheme related problems
- Distinguish between “toggled open” and “current
- Multiple “selected” items

I cannot reproduce that behavior:
- Inconsistent toggle experience

**Comment by getdave** [comment]
> Also, @getdave can you confirm that you have read my comment [#48803 (comment)](https://github.com/Automattic/wp-calypso/pull/48803#issuecomment-762134998) prior to yours [#48803 (review)](https://github.com/Automattic/wp-calypso/pull/48803#pullrequestreview-571329135)?

Not in sufficient detail. Apologies for that. **Yes** I think this needs to be addressed separately from fixing mobile UX.

**Comment by getdave** [comment]
> Multiple "selected" items
and
> Distinguish between “toggled open” and “current

Agreed. This an issue relating to those color schemes not visually differentiating between "open" and "selected". It's super confusing.

> Inconsistent toggle experience

I can no longer replicate this 👍

**Review by getdave — APPROVED** [review]
